app/Http/Requests/TransactionRequest.php

```
@@ -30,10 +36,107 @@ public function attributes(): array
+     */
+    private function byMonthRule(string $frequencyField, string $byDayField): array
+    {
+        return [
+            'nullable',
+            'integer',
+            'between:1,12',
+            // A plain closure is skipped by the validator when the field is null and
+            // 'nullable' is present, so the "required" direction needs an implicit
+            // rule (Rule::requiredIf isn't skipped) rather than a closure fail().
+            Rule::requiredIf(fn () => $this->input($frequencyField) === 'YEARLY' && (bool) $this->input($byDayField)),
+            function ($attribute, $value, $fail) use ($frequencyField) {
```

PHPMD (2.15.0)

[warning] Avoid unused parameters such as '$attribute'. (undefined)

(UnusedFormalParameter)